frontend/app/(defaultLayout)/products/[handle]/helpers.ts

```diff
@@ -0,0 +1,156 @@
+import { DEFAULT_STORE_CODE } from '@config/env';
+import { hasDisableCacheGets } from '@helpers/cache-control-helpers';
+import { getiFixitOrigin } from '@helpers/path-helpers';
+import { invariant } from '@ifixit/helpers';
+import { StoreListItem } from '@models/store';
+import type { Product } from '@pages/api/nextjs/cache/product';
+import {
+   getDefaultVariantId,
+   getSearchParamVariantId,
+} from '@templates/product/hooks/useSelectedVariant';
+import { GetServerSidePropsContext } from 'next';
+import { headers } from 'next/headers';
+import {
+   ItemAvailability as SchemaItemAvailability,
+   OfferItemCondition as SchemaOfferItemCondition,
+} from 'schema-dts';
+import { PageProps } from './page';
+
+export function hasDisableCacheGetsWrapper(
+   searchParams: Record<string, string>
+): boolean {
+   return hasDisableCacheGets({
+      query: searchParams,
+   } as GetServerSidePropsContext);
+}
+
+export function identifyStoresWithProduct(
+   product: Product,
+   stores: StoreListItem[]
+) {
+   const codeToDomain =
+      product.enabledDomains?.reduce((acc, { code, domain }) => {
+         acc[code] = domain;
+         return acc;
+      }, {} as Record<string, string>) ?? {};
+   return stores.map((store) => {
+      const domain =
+         store.code === DEFAULT_STORE_CODE
+            ? new URL(store.url).origin
+            : codeToDomain[store.code];
+      if (domain) {
+         return { ...store, url: `${domain}/products/${product.handle}` };
+      }
+      return store;
+   });
+}
+
+type CreateMetadataSupportProps = PageProps & {
+   product: Product;
+};
+
+export function createMetadataSupport({
+   params,
+   searchParams,
+   product,
+}: CreateMetadataSupportProps) {
+   type Image = typeof product.images[number];
+
+   const { handle } = params;
+   const { images, metaTitle, shortDescription } = product;
+
+   const defaultVariantId = getDefaultVariantId(product);
+   const searchParamVariantId = getSearchParamVariantId(searchParams.variant);
+   let variant = product.variants.find((v) => v.id === searchParamVariantId);
+   if (variant == null) {
+      variant = product.variants.find((v) => v.id === defaultVariantId);
+   }
```

Comment:
This bit is identical to the part in `createLdJsonSupport()` below. How about extracting this to a function `getVariant(product, searchParams)`?